src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContent.cs

```diff
@@ -202,6 +202,13 @@ public TagHelperContent SetContent(IHtmlContent htmlContent)
         /// <returns>A <see cref="string"/> containing the content.</returns>
         public abstract string GetContent();
 
+        /// <summary>
+        /// Gets the content.
+        /// </summary>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        /// <returns>A <see cref="string"/> containing the content.</returns>
+        public abstract string GetContent(IHtmlEncoder encoder);
```

Comment:
This change was made as a bit of a drive-by.

The problem with what both are you are suggesting is that the HtmlEncoder is a service and is replaceable. We either need to pass it in to the constructor, or drop the parameterless overload.

So while I agree that having `GetContent()` be virtual non-abstract here would be a more expected design, it's not really correct either because we just shouldn't provide that API.